obfuscation.go

```diff
@@ -0,0 +1,188 @@
+package sphinx
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"errors"
+	"io"
+
+	"github.com/roasbeef/btcd/btcec"
+)
+
+// onionObfuscation obfuscates the data with compliance with BOLT#4.
```

Comment:
Either a newline should be inserted, or the line below should be properly wrapped (respecting the 80 character limit) into the one above.